neuromation/cli/ael.py

```diff
@@ -0,0 +1,460 @@
+        tasks.append(
+            loop.create_task(
+                _process_resizing(
+                    functools.partial(root.client.jobs.resize, job), stdout
+                )
+            )
+        )
+        await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
+        for task in tasks:
+            await root.cancel_with_logging(task)
+
+
+async def _process_resizing(
+    resizer: Callable[..., Awaitable[None]], stdout: Output
+) -> None:
+    loop = asyncio.get_event_loop()
+    resize_event = asyncio.Event()
+
+    def resize() -> None:
+        resize_event.set()
+
+    has_sigwinch = (
+        hasattr(signal, "SIGWINCH")
+        and threading.current_thread() is threading.main_thread()
+    )
+    if has_sigwinch:
+        previous_winch_handler = signal.getsignal(signal.SIGWINCH)
+        loop.add_signal_handler(signal.SIGWINCH, resize)
+        if previous_winch_handler is None:
+            # Borrowed from the Prompt Toolkit.
+            # In some situations we receive `None`. This is
+            # however not a valid value for passing to
+            # `signal.signal` at the end of this block.
+            previous_winch_handler = signal.SIG_DFL
+
+    prevh = prevw = None
+    try:
+        while True:
+            if has_sigwinch:
+                await resize_event.wait()
+                resize_event.clear()
+            else:
+                # Windows or non-main thread
+                # The logic is borrowed from docker CLI.
+                # Wait for 250 ms
+                # If there is no resize event -- check the size anyway on timeout.
+                # It makes resizing to work on Windows.
+                await asyncio.sleep(0.25)
+            h, w = stdout.get_size()
+            if prevh != h or prevw != w:
+                prevh = h
+                prevw = w
+                await resizer(w=w, h=h)
+    finally:
+        if has_sigwinch:
+            loop.remove_signal_handler(signal.SIGWINCH)
+            signal.signal(signal.SIGWINCH, previous_winch_handler)
+
+
+async def _process_stdin_tty(stream: StdStream) -> None:
+    ev = asyncio.Event()
+
+    def read_ready() -> None:
+        ev.set()
+
+    inp = create_input()
+    with inp.raw_mode():
+        with inp.attach(read_ready):
+            while True:
+                await ev.wait()
+                ev.clear()
+                if inp.closed:
+                    return
+                keys = inp.read_keys()  # + inp.flush_keys()
+                buf = b"".join(key.data.encode("utf8") for key in keys)
+                await stream.write_in(buf)
+
+
+async def _process_stdout_tty(stream: StdStream, stdout: Output) -> None:
+    codec_info = codecs.lookup("utf8")
+    decoder = codec_info.incrementaldecoder("replace")
+    while True:
+        chunk = await stream.read_out()
+        if chunk is None:
+            return
+        txt = decoder.decode(chunk.data)
+        if txt is not None:
```

Comment:
Can it be None?

Reply:
You are right.
Removed from here and from all similar places.